modules/core/04-channel/types/msgs.go

```diff
@@ -569,29 +566,17 @@ func (msg MsgChannelUpgradeTry) ValidateBasic() error {
 	if !IsValidChannelID(msg.ChannelId) {
 		return ErrInvalidChannelIdentifier
 	}
-	if msg.CounterpartyChannel.State != INITUPGRADE {
-		return errorsmod.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", INITUPGRADE, msg.CounterpartyChannel.State)
-	}
-	if msg.CounterpartySequence == 0 {
+	if msg.CounterpartyUpgradeSequence == 0 {
 		return errorsmod.Wrap(ibcerrors.ErrInvalidSequence, "counterparty sequence cannot be 0")
 	}
-	if msg.ProposedUpgradeChannel.State != TRYUPGRADE {
-		return errorsmod.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", TRYUPGRADE, msg.CounterpartyChannel.State)
-	}
-	if msg.TimeoutHeight.IsZero() {
-		return errorsmod.Wrap(ibcerrors.ErrInvalidHeight, "timeout height must be non-zero")
-	}
-	if msg.TimeoutTimestamp == 0 && msg.TimeoutHeight.IsZero() {
-		return errorsmod.Wrap(ErrInvalidUpgradeTimeout, "invalid upgrade timeout timestamp or timeout height")
+	if msg.ProposedUpgradeTimeout.Height.IsZero() && msg.ProposedUpgradeTimeout.Timestamp == 0 {
+		return errorsmod.Wrap(ErrInvalidUpgradeTimeout, "timeout height or timeout timestamp must be non-zero")
```

Comment:
I would order checks based on their proto ordering, so this could come before the counterparty upgrade sequence